core/arch/arm/tee/entry_fast.c

```diff
@@ -51,31 +51,75 @@ static void tee_entry_fastcall_l2cc_mutex(struct thread_smc_args *args)
 
 #ifdef ARM32
 	switch (args->a1) {
-	case TEESMC_OPTEE_L2CC_MUTEX_GET_ADDR:
+	case OPTEE_SMC_L2CC_MUTEX_GET_ADDR:
 		ret = tee_get_l2cc_mutex(&args->a2);
 		break;
-	case TEESMC_OPTEE_L2CC_MUTEX_SET_ADDR:
+	case OPTEE_SMC_L2CC_MUTEX_SET_ADDR:
 		ret = tee_set_l2cc_mutex(&args->a2);
 		break;
-	case TEESMC_OPTEE_L2CC_MUTEX_ENABLE:
+	case OPTEE_SMC_L2CC_MUTEX_ENABLE:
 		ret = tee_enable_l2cc_mutex();
 		break;
-	case TEESMC_OPTEE_L2CC_MUTEX_DISABLE:
+	case OPTEE_SMC_L2CC_MUTEX_DISABLE:
 		ret = tee_disable_l2cc_mutex();
 		break;
 	default:
-		args->a0 = TEESMC_RETURN_EBADCMD;
+		args->a0 = OPTEE_SMC_RETURN_EBADCMD;
 		return;
 	}
 #else
 	ret = TEE_ERROR_NOT_SUPPORTED;
 #endif
 	if (ret == TEE_ERROR_NOT_SUPPORTED)
-		args->a0 = TEESMC_RETURN_UNKNOWN_FUNCTION;
+		args->a0 = OPTEE_SMC_RETURN_UNKNOWN_FUNCTION;
 	else if (ret)
-		args->a0 = TEESMC_RETURN_EBADADDR;
+		args->a0 = OPTEE_SMC_RETURN_EBADADDR;
 	else
-		args->a0 = TEESMC_RETURN_OK;
+		args->a0 = OPTEE_SMC_RETURN_OK;
+}
+
+static void tee_entry_exchange_capabilities(struct thread_smc_args *args)
+{
+	if (args->a1) {
+		/*
+		 * Either unknown capability or
+		 * OPTEE_SMC_NSEC_CAP_UNIPROCESSOR, in either case we can't
```

Comment:
I see, then perhaps a more specific capability name would be better. Or, since `OPTEE_SMC_NSEC_CAP_UNIPROCESSOR` is not used anywhere yet, why define it now?